pkg/app/api/grpcapi/web_api.go

```diff
@@ -1375,3 +1384,25 @@ func (a *WebAPI) GetInsightData(ctx context.Context, req *webservice.GetInsightD
 		DataPoints: idp,
 	}, nil
 }
+
+func (a *WebAPI) getInsightFromCache(projectID, appID string, kind model.InsightMetricsKind, step model.InsightStep, from time.Time, count int) (insightstore.Chunks, error) {
+	var chunks insightstore.Chunks
+	paths := insightstore.DetermineFilePaths(projectID, appID, kind, step, from, count)
+	for _, p := range paths {
+		c, err := a.insightCache.Get(p)
+		chunk, ok := c.(insightstore.Chunk)
+		if err != nil || !ok {
+			return nil, fmt.Errorf("failed to get insight from cache")
+		}
+		chunks = append(chunks, chunk)
+	}
+
+	return chunks, nil
+}
+
+func (a *WebAPI) putInsightToCache(chunks insightstore.Chunks) {
+	for _, c := range chunks {
+		a.insightCache.Put(c.GetFilePath(), c)
+	}
+	return
+}
```

Comment:
Thank you. I am thinking about moving this part to outside of this file. For example,
- changing the package name of `insightstore` to `insight` (because it contains `Store`, `Cache` and `Datapoint` struct)
- moving this part into the `insight` package as: `LoadChunksCache(c cache.Cache, ...)`, `PutChunksToCache(c cache.Cache, ...)`

How do you think?

Reply:
Sure, I agree with you.